snomed/com.b2international.snowowl.validation.snomed/src/main/resources/scripts/rule664.groovy

```diff
@@ -39,23 +39,24 @@ ExpressionBuilder activeFsnExpression = Expressions.builder()
 		.filter(SnomedDescriptionIndexEntry.Expressions.type(Concepts.FULLY_SPECIFIED_NAME))
 		.filter(SnomedDescriptionIndexEntry.Expressions.concepts(activeConceptIds))
 
-if (params.isUnpublishedOnly) {
-	activeFsnExpression.filter(SnomedDocument.Expressions.effectiveTime(EffectiveTimes.UNSET_EFFECTIVE_TIME))
-}
-
-Aggregation<String> activeDescriptionsByOriginalTerm = searcher
-		.aggregate(AggregationBuilder.bucket("ruleSnomedCommon2", String.class, SnomedDescriptionIndexEntry.class)
+Aggregation<String[]> activeDescriptionsByOriginalTerm = searcher
+		.aggregate(AggregationBuilder.bucket("ruleSnomedCommon2", String[].class, SnomedDescriptionIndexEntry.class)
 		.query(activeFsnExpression.build())
 		.onFieldValue(SnomedDescriptionIndexEntry.Fields.ORIGINAL_TERM)
-		.fields(SnomedDescriptionIndexEntry.Fields.ID)
+		.fields(SnomedDescriptionIndexEntry.Fields.ID,
+			SnomedDescriptionIndexEntry.Fields.EFFECTIVE_TIME)
 		.minBucketSize(2))
 
-		
 List<ComponentIdentifier> issues = Lists.newArrayList()
 
 activeDescriptionsByOriginalTerm.getBuckets().values().each({ bucket ->
-	bucket.each({ id ->
-		issues.add(ComponentIdentifier.of(SnomedTerminologyComponentConstants.DESCRIPTION_NUMBER, id))
+	bucket.each({ hit ->
+		def id = hit[0]
+		def effectiveTime = hit[1]
+		
+		if (!params.isUnpublishedOnly || Long.valueOf(effectiveTime).equals(EffectiveTimes.UNSET_EFFECTIVE_TIME)) {
```

Comment:
Hint: this can be used to check if a value is an unset effective time: `com.b2international.snowowl.core.date.EffectiveTimes.isUnset(Object)`.